graylog2-server/src/main/java/org/graylog2/indexer/IndexMapping5.java

```diff
@@ -0,0 +1,72 @@
+/**
+ * This file is part of Graylog.
+ *
+ * Graylog is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * Graylog is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with Graylog.  If not, see <http://www.gnu.org/licenses/>.
+ */
+package org.graylog2.indexer;
+
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableMap;
+
+import javax.inject.Singleton;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * Representing the message type mapping in Elasticsearch 5.x. This is giving ES more
+ * information about what the fields look like and how it should analyze them.
+ *
+ * @see <a href="https://www.elastic.co/guide/en/elasticsearch/reference/5.4/mapping.html">Elasticsearch Reference / Mapping</a>
+ */
+@Singleton
+public class IndexMapping5 extends IndexMapping {
+    protected List<Map<String, Map<String, Object>>> dynamicTemplate() {
+        final Map<String, Object> defaultInternal = ImmutableMap.of(
+                "match", "gl2_*",
+                "mapping", notAnalyzedString());
+        final Map<String, Map<String, Object>> templateInternal = ImmutableMap.of("internal_fields", defaultInternal);
+
+        final Map<String, Object> defaultAll = ImmutableMap.of(
+                // Match all
+                "match", "*",
+                // Analyze nothing by default
+                "mapping", ImmutableMap.of("index", "not_analyzed"));
+        final Map<String, Map<String, Object>> templateAll = ImmutableMap.of("store_generic", defaultAll);
+
+        return ImmutableList.of(templateInternal, templateAll);
+    }
```

Comment:
This method is an exact duplicate of the one in [`IndexMapping2`](https://github.com/Graylog2/graylog2-server/blob/es-http-index-mapping/graylog2-server/src/main/java/org/graylog2/indexer/IndexMapping2.java#L35-L49). I would suggest we include it in the base class and make it abstract when we are really overriding it in a subclass.